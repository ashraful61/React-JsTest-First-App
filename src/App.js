import { useState } from 'react';
import './App.css';

function App() {

  const [data, setData] = useState("")

  return (
    <div className="App">
      <p>my test react app</p>
      
      {/* <input type='text' placeholder='user name' name="user_name" /> <br /> */}
      
      <h1>Snapshot Testing</h1>
      <p>{data}</p>

      {/* <input type='text' placeholder='full name' name="full_name" onChange={ e => setData(e.target.value)}/> */}
      <button onClick={ e => setData('Updated data test')}>Update data</button>

      <br/>
      <br/>
      <br/>
      <br/>

      <img title='test image title' src="https://api.deepai.org/job-view-file/e67717d4-f9c0-4b32-b1f7-c94e38cf1328/outputs/output.jpg" alt="" />
   
    </div>
  );
}

export default App;
