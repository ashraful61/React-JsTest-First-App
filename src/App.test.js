import { act, fireEvent, render, screen } from '@testing-library/react';
import App from './App';

// describe('app component testing', () => {

//   test('first react app test case 1', () => {
//     render(<App />);
//     const text = screen.getByText("my test react app");
//     expect(text).toBeInTheDocument();
//     const imgTitle = screen.getByTitle("test image title");
//     expect(imgTitle).toBeInTheDocument();

//   });

//   test('input box testing test case 2', () => {
//     render(<App />);
//     let checkInput = screen.getByRole("textbox");
//     expect(checkInput).toBeInTheDocument();

//     const getPlaceHolderText = screen.getByPlaceholderText("user name");
//     expect(getPlaceHolderText).toBeInTheDocument();

//     expect(checkInput).toHaveAttribute('name', 'user_name')
//   });

// })

// describe.skip('app component testing 2', () => {

//   test('description 2', () => {
//     render(<App />);
//     const text = screen.queryByText("aasdas as");
//     expect(text).not.toBeInTheDocument();
//   });

// })

// describe.only('app component testing 3', () => {

//   test('description 2', () => {
//     render(<App />);
//     const text = screen.queryByText("aasdas as");
//     expect(text).not.toBeInTheDocument();
//   });

// })



// describe('app component with Input change event', () => {

//   test('description 2', () => {
//     render(<App />);
//     const text = screen.queryByText("aasdas as");
//     expect(text).not.toBeInTheDocument();
//   });

// })

// test("On change event testing", () => {
//   render(<App />);

//   const input = screen.getByRole("textbox");

//   fireEvent.change(input, {target: {value: "a"}})
//   expect(input.value).toBe('a')

// })
test("click event testing", () => {
  render(<App />);

  const btn = screen.getByRole("button");
  fireEvent.click(btn);
  expect(screen.getByText('Updated data test')).toBeInTheDocument()

})


// Snapshot testing 


test('snapshot for app component', async () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });


