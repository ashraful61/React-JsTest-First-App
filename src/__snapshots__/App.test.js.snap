// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for app component 1`] = `
<div>
  <div
    class="App"
  >
    <p>
      my test react app
    </p>
    <h1>
      Snapshot Testing
    </h1>
    <p />
    <button>
      Update data
    </button>
    <br />
    <br />
    <br />
    <br />
    <img
      alt=""
      src="https://api.deepai.org/job-view-file/e67717d4-f9c0-4b32-b1f7-c94e38cf1328/outputs/output.jpg"
      title="test image title"
    />
  </div>
</div>
`;
